irohad/model/generators/impl/transaction_generator.cpp

```diff
@@ -33,25 +34,29 @@ namespace iroha {
         CommandGenerator command_generator;
         // Add peers
         for (size_t i = 0; i < peers_address.size(); ++i) {
-          // TODO: replace with more flexible scheme, generate public keys with
-          // specified parameters
-          auto peer_key = generator::random_blob<pubkey_t::size()>(i + 1);
-          tx.commands.push_back(
-              command_generator.generateAddPeer(peers_address[i], peer_key));
+          KeysManagerImpl manager("node" + std::to_string(i));
+          manager.createKeys("node" + std::to_string(i));
+          auto keypair = *manager.loadKeys();
```

Comment:
seems sophisticated

Reply:
Current case is a bit unusual since we need to generate keys first. In most cases 2 calls enough (ctor + loadKeys).